### ISSUE TITLE
[Bug] DBFC & DFCI: Scoreboard fails to update after ranked match

### ISSUE THREAD

**Issue by Ninetime** [RPCS3/rpcs3#15824]
### Quick summary

When playing Dengeki Bunko: Fighting Climax (DBFC) & Dengeki Bunko: Fighting Climax Ignition (DFCI) the leaderboards fails to update after a ranked match. Afterwards, if you attempt to go to the Ranking section, it will throw and error and send you to the main screen.

### Details

After completing a ranked match, the following error is thrown:
```
E {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd199c, LR:0x000efb68]} SYS: 'sceNpScoreDestroyTransactionCtx' failed with 0x8002a1a3 : SCE_NP_COMMUNITY_ERROR_INVALID_ID [1]
W {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd199c, LR:0x000efb68]} sceNp: sceNpScoreDestroyTransactionCtx(transId=4098)
E {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd199c, LR:0x000efb68]} SYS: 'sceNpScoreDestroyTransactionCtx' failed with 0x8002a1a3 : SCE_NP_COMMUNITY_ERROR_INVALID_ID [2]
W {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd199c, LR:0x000efb68]} sceNp: sceNpScoreDestroyTransactionCtx(transId=4099)
E {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd199c, LR:0x000efb68]} SYS: 'sceNpScoreDestroyTransactionCtx' failed with 0x8002a1a3 : SCE_NP_COMMUNITY_ERROR_INVALID_ID [3]
```

Afterwards, attempting to access Ranking results in the following:
![BLUS31535_screenshot_2024_07_22_18_26_24](https://github.com/user-attachments/assets/c68b33b3-6a52-4eeb-8f43-283a99a44013)

### Attach a log file

[RPCS3.log.gz](https://github.com/user-attachments/files/16340254/RPCS3.log.gz)

### System configuration
AMD Ryzen 7 5800X
31.92 GiB RAM
AMD Radeon RX 6800 XT (24.2.1)
Windows 10 22H2
RPCS3 v0.0.32-16705-c533dc96

**Comment by Ninetime** [comment]
From reviewing the logs further, it seems the old room id is nulled/set to 0 after leaving the room. The issue being the game calls `sceNpMatching2SignalingGetConnectionInfo` afterwards and 0 is an invalid ID.
 ```
·W 0:16:42.549469 {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd62d4, LR:0x000ec660]} sceNp2: sceNpMatching2SignalingGetConnectionInfo(ctxId=1, roomId=3, memberId=33, code=6, connInfo=*0xd00406b8)
·W 0:16:45.965944 {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd62bc, LR:0x000eacc0]} sceNp2: sceNpMatching2LeaveRoom(ctxId=1, reqParam=*0xd0040480, optParam=*0xd0040470, assignedReqId=*0x78a420)
·W 0:16:46.174675 {NP Handler Thread} sceNp2: SceNpMatching2RoomMemberDataInternal:
·W 0:16:46.174695 {NP Handler Thread} sceNp2: next: *0x0
·W 0:16:46.174714 {NP Handler Thread} sceNp2: npId: Ninetimejp
·W 0:16:46.174732 {NP Handler Thread} sceNp2: onlineName: Ninetimejp
·W 0:16:46.174751 {NP Handler Thread} sceNp2: avatarUrl: https://rpcs3.net/cdn/netplay/DefaultAvatar.png
·W 0:16:46.174770 {NP Handler Thread} sceNp2: joinDate: 63863075482090737
·W 0:16:46.174788 {NP Handler Thread} sceNp2: memberId: 33
·W 0:16:46.174807 {NP Handler Thread} sceNp2: teamId: 0
·W 0:16:46.174825 {NP Handler Thread} sceNp2: roomGroup: *0x0
·W 0:16:46.174844 {NP Handler Thread} sceNp2: natType: 2
·W 0:16:46.174862 {NP Handler Thread} sceNp2: flagAttr: 0x0
·W 0:16:46.174881 {NP Handler Thread} sceNp2: roomMemberBinAttrInternal: *0x7aa7c0
·W 0:16:46.174899 {NP Handler Thread} sceNp2: roomMemberBinAttrInternalNum: 1
·W 0:16:46.174918 {NP Handler Thread} sceNp2: updateDate: 63863075482090736
·W 0:16:46.174937 {NP Handler Thread} sceNp2: Id: 89, Size: 0, ptr: *0x7aa7d8
·! 0:16:46.182947 {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd630c, LR:0x000e7ee8]} sceNp2: sceNpMatching2GetEventData(ctxId=1, eventKey=10, buf=*0x32312960, bufLen=312)
·W 0:16:46.815916 {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd62d4, LR:0x000ec608]} sceNp2: sceNpMatching2SignalingGetConnectionInfo(ctxId=1, roomId=0, memberId=33, code=1, connInfo=*0xd0040670)
·E 0:16:46.815994 {PPU[0x1000000] Thread (main_thread) [HLE:0x00bd62d4, LR:0x000ec608]} SYS: 'sceNpMatching2SignalingGetConnectionInfo' failed with 0x8002230f : SCE_NP_MATCHING2_ERROR_INVALID_ROOM_ID [1]
```

**Comment by Ninetime** [comment]
Resolved by #16152
